app/src/main/java/com/github/houseorganizer/houseorganizer/shop/FirestoreShopList.java

```diff
@@ -0,0 +1,154 @@
+            return snap;
+        });
+    }
+
+
+
+    private static List<Map<String, Object>> convertItemsListToFirebase(List<ShopItem> shopItemList){
+        List<Map<String, Object>> items = new LinkedList<>();
+
+        for(int i = 0; i < shopItemList.size(); ++i){
+            ShopItem item = shopItemList.get(i);
+            Map<String, Object> itemMap = new HashMap<>();
+            itemMap.put("name", item.getName());
+            itemMap.put("quantity", item.getQuantity());
+            itemMap.put("unit", item.getUnit());
+            itemMap.put("pickedUp", item.isPickedUp());
+            items.add(itemMap);
+        }
+        return items;
+    }
+
+    private static List<ShopItem> convertFirebaseListToItems(List<Map<String, Object>> list){
+        List<ShopItem> items = new LinkedList<>();
+        for(Map<String, Object> m : list){
+            items.add(new ShopItem((String)m.get("name"),
+                    new Long((long) m.get("quantity")).intValue(),
+                    (String) m.get("unit")));
+            items.get(items.size()-1).setPickedUp((boolean) m.get("pickedUp"));
+        }
+        return items;
+    }
+
+    /**
+     * Add a shopList for a new house : DOES NOT CHECK if a house already have a list
+     * @param shopListRoot root folder of shop lists
+     * @param shopList shopList to add
+     * @param household household linked to the grocery list
+     * @return the adding task with as result the new document
+     */
+    public static Task<DocumentReference> storeNewShopList(CollectionReference shopListRoot, ShopList shopList, DocumentReference household){
+        Map<String, Object> map = new HashMap<>();
+        map.put("household", household);
+        List<Map<String, Object>> items = convertItemsListToFirebase(shopList.getItems());
+        map.put("items", items);
+        return shopListRoot.add(map);
+    }
+
+
+    public static Task<FirestoreShopList> retrieveShopList(CollectionReference shopListRoot, DocumentReference household){
+        return shopListRoot.whereEqualTo("household", household).get().continueWith( t -> {
+            List<DocumentSnapshot> res = t.getResult().getDocuments();
+            if(res.isEmpty())return null;
+            if(res.size() > 1) throw new IllegalStateException("More than one groceries list for this house");
+            return buildShopList(res.get(0));
+        });
+    }
+
+    public static FirestoreShopList buildShopList(DocumentSnapshot documentSnapshot){
+        if(documentSnapshot == null) return null;
+        DocumentReference household = (DocumentReference) documentSnapshot.get("household");
+        List<Map<String, Object>> list = (List<Map<String, Object>>) documentSnapshot.get("items");
+        List<ShopItem> items = convertFirebaseListToItems(list);
+        return new FirestoreShopList(household, documentSnapshot.getReference(), items);
+    }
+}
```

Comment:
Is it necessary to have both a FireStore shoplist and a non FireStore one ? I feel like only one would suffice

Reply:
I think he wanted to keep the model separate from the implementation-specific use of Firebase

Reply:
Yes exactly but I admit that the New version of shoplist is very close to à simple list which was not the case before